ios/FluentUI/Vnext/ActivityIndicator/ActivityIndicator.swift

```diff
@@ -3,39 +3,63 @@
-    var isAnimating: Bool { get set }
-    var hidesWhenStopped: Bool { get set }
-    var size: MSFActivityIndicatorSize { get set }
-}
+/// UIKit wrapper that exposes the SwiftUI Activity Indicator implementation
+@objc open class MSFActivityIndicator: NSObject, FluentUIWindowProvider {
 
-/// Properties available to customize the state of the activity indicator
-class MSFActivityIndicatorStateImpl: NSObject, ObservableObject, MSFActivityIndicatorState {
-    @Published var color: UIColor?
-    @Published var isAnimating: Bool = false
-    @Published var hidesWhenStopped: Bool = true
+    /// The UIView representing the Activity Indicator.
+    @objc open var view: UIView {
+        return hostingController.view
+    }
 
-    let tokens: MSFActivityIndicatorTokens
+    /// The object that groups properties that allow control over the  Activity Indicator appearance.
+    @objc open var state: MSFActivityIndicatorState {
+        return self.activityIndicatorView.state
+    }
 
-    var size: MSFActivityIndicatorSize {
-        get {
-            return tokens.size
-        }
+    /// Creates a new MSFActivityIndicator instance.
+    /// - Parameter size: The MSFActivityIndicatorSize value used by the Activity Indicator.
+    @objc public convenience init(size: MSFActivityIndicatorSize = .medium) {
+        self.init(size: size,
+                  theme: nil)
+    }
 
-        set {
-            tokens.size = newValue
-        }
+    /// Creates a new MSFActivityIndicator instance.
+    /// - Parameters:
+    ///   - size: The MSFActivityIndicatorSize value used by the Activity Indicator.
+    ///   - theme: The FluentUIStyle instance representing the theme to be overriden for this Activity Indicator.
+    @objc public init(size: MSFActivityIndicatorSize = .medium,
+                      theme: FluentUIStyle? = nil) {
+        super.init()
+
+        activityIndicatorView = ActivityIndicator(size: size)
+        hostingController = UIHostingController(rootView: AnyView(activityIndicatorView
+                                                                    .windowProvider(self)
+                                                                    .modifyIf(theme != nil, { activityIndicatorView in
+                                                                        activityIndicatorView.customTheme(theme!)
+                                                                    })))
+        hostingController.disableSafeAreaInsets()
+        view.backgroundColor = UIColor.clear
     }
 
-    init(size: MSFActivityIndicatorSize) {
-        self.tokens = MSFActivityIndicatorTokens(size: size)
+    var window: UIWindow? {
+        return self.view.window
     }
+
+    private var hostingController: UIHostingController<AnyView>!
+
+    private var activityIndicatorView: ActivityIndicator!
+}
+
+/// Properties available to customize the state of the activity indicator state
+@objc public protocol MSFActivityIndicatorState {
```

Comment:
Could you document the public properties on this protocol?

Reply:
Sure! Missed these and the View struct initializer too.